payments/admin.py

```diff
@@ -103,15 +103,15 @@ def pretax_price(self, obj):
 
     @currency
     def total_price(self, obj):
-        return obj.total_price
+        return obj.total_price if obj.price and obj.tax_percentage else None
```

Comment:
The problem we have in Django admin is that Django generate an empty order line inline (even though we specify `extra = 0`), which of course has no prices. Once that inline tries to populate `pretax_price` (lines 68-70 in this file), decorator `currency` fails. I think that the changes you added now to the `currency_format` and `percentage_format` are good and what we need here. However, all these `if obj.price and obj.tax_percentage`, I feel, should not be here. We should ensure that all the order and order line properties do not fail and either return the value or return `None`, but do not end up in errors.

Reply:
The reason for those `if obj.price` is different than the one that was making `@currency` fail.
```
Exception Value: unsupported operand type(s) for /: 'NoneType' and 'int'
Exception Location: /berth-reservations/payments/utils.py in convert_aftertax_to_pretax, line 55
```

Since `total_price` is calculated by `sum([ol.price for ol in self.order_lines.all()], self.price)`, the `self.price == None` for those empty orders being generated on the admin. With those checks we ensure that the `total_price` is not being calculated _if_ the order doesn't have a price.

There are two alternatives:
1. Leave it as-is, since those checks only concern the admin, which is the only place where "empty" orders can be created without a price
2. Add a some kind of validation to
```python\

Reply:
yep, you are right, after further investigation also seemed like these were needed, let's leave them then 👍